### DISCUSSION TITLE
error prmisson after transfer dms in new server (copy paste)

### DISCUSSION THREAD

**Discussion by vahidalvandi** [docker-mailserver/docker-mailserver#3986] [General]
hi.  I got an unexpected error that I had to change the access level of the folder to 777 but I got the following error to postmaster daily

```
/etc/cron.daily/logrotate:
error: skipping "/var/log/mail/clamav.log" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/freshclam.log" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/fail2ban.log" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/mail.log" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/mail.info" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/mail.warn" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
error: skipping "/var/log/mail/mail.err" because parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell logrotate which user/group should be used for rotation.
run-parts: /etc/cron.daily/logrotate exited with return code 1
```

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The error messages you are seeing indicate that the permissions for the `/var/log/mail` directory are too permissive, which is causing logrotate to skip the rotation of the log files. Setting the permissions to `777` (world writable) is generally not recommended due to security concerns, especially for directories that contain sensitive log files.

To resolve this issue, you should adjust the permissions of the `/var/log/mail` directory to be more restrictive and owned by the root user and group. Here are the steps you can follow:

1. Change the ownership of the directory to root:
   ```bash
   sudo chown root:root /var/log/mail
   ```

2. Change the permissions of the directory to allow only the owner (root) to write to it:
   ```bash
   sudo chmod 755 /var/log/mail
   ```

3. Optionally, you can specify the user and group that logrotate should use when rotating logs by adding the `su` directive in the logrotate configuration file for your mail logs. This is typically located in `/etc/logrotate.d/` or directly in the logrotate configuration for the mail service. You would add:
   ```
   su root root
   ```
   This tells logrotate to perform log rotation operations as the root user and group.

After making these changes, logrotate should be able to perform the log rotation without encountering the permissions error. If you continue to experience issues, ensure that no other configurations are overriding these settings or that there are no additional security policies affecting these operations.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by vahidalvandi** [comment]
thank you
